tests/device_tests/test_msg_resetdevice_supershamir.py

```diff
@@ -0,0 +1,232 @@
+        member_threshold = 3
+
+        def input_flow():
+            # Confirm Reset
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # Backup your seed
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # Confirm warning
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # shares info
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # Set & Confirm number of groups
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # threshold info
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # Set & confirm group threshold value
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            for _ in range(5):
+                # Set & Confirm number of share
+                btn_code = yield
+                assert btn_code == B.ResetDevice
+                self.client.debug.press_yes()
+
+                # Set & confirm share threshold value
+                btn_code = yield
+                assert btn_code == B.ResetDevice
+                self.client.debug.press_yes()
+
+            # Confirm show seeds
+            btn_code = yield
+            assert btn_code == B.ResetDevice
+            self.client.debug.press_yes()
+
+            # show & confirm shares for all groups
+            all_mnemonics = []
+            for g in range(5):
+                for h in range(5):
+                    words = []
+                    btn_code = yield
+                    assert btn_code == B.Other
+
+                    # mnemonic phrases
+                    # 20 word over 6 pages for strength 128, 33 words over 9 pages for strength 256
+                    for i in range(6):
+                        words.extend(self.client.debug.read_reset_word().split())
+                        if i < 5:
+                            self.client.debug.swipe_down()
+                        else:
+                            # last page is confirmation
+                            self.client.debug.press_yes()
+
+                    # check share
+                    for _ in range(3):
+                        index = self.client.debug.read_reset_word_pos()
+                        self.client.debug.input(words[index])
+
+                    all_mnemonics.extend([" ".join(words)])
+
+                    # Confirm continue to next share
+                    btn_code = yield
+                    assert btn_code == B.Success
+                    self.client.debug.press_yes()
+
+            # generate secret locally
```

Comment:
is there a reason for this part to be in input_flow? it should be in the test function, after the device.reset call, IMHO

Reply:
probably not, it was done the same way in resetdevice_shamir. Fixed in both.